### PR TITLE
release: v1.9.0

### PR DESCRIPTION
**Pull request by claytonparnell**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by claytonparnell** [comment]
### Testing

```
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_cpu[keras.test.Dockerfile-required_packages0] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_cpu[scipy.test.Dockerfile-required_packages4] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_cpu[pandas.test.Dockerfile-required_packages7] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_cpu[sm-python-sdk.test.Dockerfile-required_packages8] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_cpu[pytorch.examples.Dockerfile-required_packages9] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[keras.test.Dockerfile-required_packages0] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[scipy.test.Dockerfile-required_packages4] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[sm-python-sdk.test.Dockerfile-required_packages8] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[pytorch.examples.Dockerfile-required_packages9] - assert 1 == 0
```
* **keras**: Documented [here](https://github.com/aws/sagemaker-distribution/issues/413). we pin to an old commit in test which works with 2.12, but not with 2.14;  ran test locally without pin and it succeeds. Will fix in followup to dynamically get correct tests.
* **scipy**: Documented [here](https://github.com/aws/sagemaker-distribution/issues/30): Tests fail with `E   TypeError: 'dia_matrix' object is not subscriptable` or `E   TypeError: 'coo_matrix' object is not subscriptable`
* **pytorch**: Several failures all related to file not found, running locally found that we're missing `sudo apt install libsdl2-dev libsdl2-2.0-0 -y` which is now needed for tests. After adding this tests succeed.
* **pandas**: Documented [here](https://github.com/aws/sagemaker-distribution/issues/110):

* **sm-python-sdk`**: Documented [here](https://github.com/aws/sagemaker-distribution/issues/316): Each test failed with either 
  * `FileNotFoundError: [Errno 2] No such file or directory: 'docker'` - expected; we don't have docker client in testing environment
  * `botocore.exceptions.NoCredentialsError: Unable to locate credentials` - also expected; we don't have creds in testing environment

File: build_artifacts/v1/v1.9/v1.9.0/gpu.env.in [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+conda-forge::sagemaker-code-editor[version='>=1.1.0,<2.0.0']
+conda-forge::amazon-sagemaker-jupyter-ai-q-developer[version='>=1.0.3,<2.0']
+conda-forge::amazon-q-developer-jupyterlab-ext[version='>=3.1.0,<4.0']
+conda-forge::amazon_sagemaker_sql_editor[version='>=0.1.7,<1.0.0']
+conda-forge::amazon-sagemaker-sql-magic[version='>=0.1.1,<1.0.0']
+conda-forge::langchain[version='>=0.1.9,<1.0.0']
+conda-forge::fastapi[version='>=0.110.3,<1.0.0']
+conda-forge::uvicorn[version='>=0.30.1,<1.0.0']
+conda-forge::pytorch-gpu[version='>=2.0.0,<3.0.0']
+conda-forge::tensorflow-gpu[version='>=2.14.0,<3.0.0']
```
Comment (balajisankar15) [inline build_artifacts/v1/v1.9/v1.9.0/gpu.env.in:13]: Why did we make this change ?
Is tensorflow-gpu officially maintained ?

**Comment by balajisankar15** [comment]
GPU and CPU variants have different versions of tensorflow. We can keep them consistent at 2.14.0

**Comment by claytonparnell** [comment]
We calculate the closure separately so we can get the highest versions, given the constraints. GPU has additional constraint but I don't think we need to dock CPU because of that. Customers are used to 2.15 and we should keep it wherever possible.

File: build_artifacts/v1/v1.9/v1.9.0/gpu.env.in (hunk 1 above)
Comment (claytonparnell) [reply build_artifacts/v1/v1.9/v1.9.0/gpu.env.in:13]: Yes it's built from the same feedstock as tensorflow (tensorflow-feedstock) and holds gpu-specific builds

**Review by balajisankar15 — APPROVED** [review]

**Review by TRNWWZ — APPROVED** [review]

**Pull request merged**